### ISSUE TITLE
Add some signal for RayCast2D/3D

### ISSUE THREAD

**Issue by CsloudX** [godotengine/godot-proposals#10606]
### Describe the project you are working on

2d platform game

### Describe the problem or limitation you are having in your project

i want check the player was in a laser shoot line. so i used a RayCast2D node.
but I found RayCast2D have no any signals for some thing collide.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add this singals for RayCast2D/RayCast3D just like Area node:

- [ ] area_entered(area)
- [ ] area_exited(area)
- [ ] body_entered(body)
- [ ] body_exited(body)

IMO, the RayCast node was same as Area node. the only diff was it was used a ray.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

none

### If this enhancement will not be used often, can it be worked around with a few lines of script?

if no signals. i think 2 way to solve this:
1. use a area instead.
2. in process or timer contious check.

but i think 2 ways both not good. for some situation, use the raycast was make sense.

### Is there a reason why this should be core and not an add-on in the asset library?

it sould be core.

**Comment by timothyqiu** [comment]
> IMO, the RayCast node was same as Area node. the only diff was it was used a ray.

No, they are not the same. A ray cast stops at the first object hit.

Therefore, it would be a single `collider_changed` signal instead of four different enter/exit signals.

> i want check the player was in a laser shoot line

Since you want to do a **check** proactively, isn't it better to call `is_colliding()`? A signal is for reacting to some event.

**Comment by Mickeon** [comment]
This proposal is similar to https://github.com/godotengine/godot/pull/95980's implementation which adds a single `collider_changed` signal. See [my comment on that implementation](https://github.com/godotengine/godot/pull/95980#issuecomment-2307409672). In my opinion, _that_ would be no better than checking for collisions every physics frame.

Adding the 4 signals is not enough, as it would be really bad for performance, so [contact_monitor](https://docs.godotengine.org/en/stable/classes/class_rigidbody2d.html#class-rigidbody2d-property-contact-monitor) would need to be added to the raycast nodes, as well.
However, as @timothyqiu mentioned, raycasts just don't work the same way as areas or physics bodies. Not that it's impossible to have them be more similar to areas, mind you.

**Comment by CsloudX** [comment]
@Mickeon YES,
like https://github.com/godotengine/godot/pull/95980 was i want
